hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/CustomFeeAssessmentStep.java

```diff
@@ -167,41 +168,50 @@ private CryptoTransferTransactionBody changedInputTxn(
         final var copy = op.copyBuilder();
         final var changedFungibleTokenTransfers = result.getMutableInputTokenAdjustments();
         final List<TokenTransferList> tokenTransferLists = new ArrayList<>();
-        // If there are no changes for the token , add as it is
-        for (final var xfers : op.tokenTransfers()) {
+        for (final var xfers : op.tokenTransfersOrElse(emptyList())) {
             final var token = xfers.token();
+            // If there are no changes for the token, leave its list untouched
             if (!changedFungibleTokenTransfers.containsKey(token)) {
                 tokenTransferLists.add(xfers);
+            } else {
+                final var postAssessmentBalances = changedFungibleTokenTransfers.get(token);
+                final var includedNetNewChanges =
+                        postAssessmentBalances.size() > xfers.transfersOrThrow().size();
```

Comment:
We have this `xfers.transfersOrThrow()` a few times in this block.  It would be good to extract that to a variable at the beginning of the block.

Reply:
Done.